### PR TITLE
[docs] Document using a different separator for read_csv

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

See discussion in https://github.com/ray-project/ray/issues/27738

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/read_api.py [hunk 1]
```diff
@@ -562,6 +562,15 @@ def read_csv(
         >>> ray.data.read_csv( # doctest: +SKIP
         ...     ["s3://bucket/path1", "s3://bucket/path2"])
 
+        >>> # Use different delimiter. The partition_filter=None is needed here
+        >>> # because by default read_csv only reads .csv files.
+        >>> from pyarrow import csv
+        >>> parse_options = csv.ParseOptions(delimiter="\\t")
+        >>> ray.data.read_csv( # doctest: +SKIP
```
Comment (pcmoritz) [inline python/ray/data/read_api.py:569]: This needs to be skipped at the moment which is very very sad -- tracked in https://github.com/ray-project/ray/issues/27853

File: python/ray/data/read_api.py [hunk 2]
```diff
@@ -562,6 +562,15 @@ def read_csv(
         >>> ray.data.read_csv( # doctest: +SKIP
         ...     ["s3://bucket/path1", "s3://bucket/path2"])
 
+        >>> # Use different delimiter. The partition_filter=None is needed here
```
Comment (matthewdeng) [inline python/ray/data/read_api.py:565]: nit: For consistency with other examples

```suggestion
        >>> # Read files that use a different delimiter. The partition_filter=None is needed here
```

**Review by matthewdeng — APPROVED** [review]
@c21 take a look?

File: python/ray/data/read_api.py [hunk 3]
```diff
@@ -562,6 +562,15 @@ def read_csv(
         >>> ray.data.read_csv( # doctest: +SKIP
         ...     ["s3://bucket/path1", "s3://bucket/path2"])
 
+        >>> # Read files that use a different delimiter. The partition_filter=None is needed here
+        >>> # because by default read_csv only reads .csv files.
+        >>> from pyarrow import csv
+        >>> parse_options = csv.ParseOptions(delimiter="\\t")
```
Comment (c21) [inline python/ray/data/read_api.py:568]: it should be `delimiter="\t"` right? I tested it out in https://github.com/ray-project/ray/issues/27738#issuecomment-1211155880 .

File: python/ray/data/examples/data/iris.tsv [hunk 4]
```diff
@@ -0,0 +1,150 @@
+5.1	3.5	1.4	0.2	setosa
```
Comment (c21) [inline python/ray/data/examples/data/iris.tsv:1]: We also need to define the schema at the first line:

```
sepal.length sepal.width petal.length petal.width variety
```

File: python/ray/data/examples/data/iris.tsv [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+6.7	3.1	5.6	2.4	virginica
+6.9	3.1	5.1	2.3	virginica
+5.8	2.7	5.1	1.9	virginica
+6.8	3.2	5.9	2.3	virginica
+6.7	3.3	5.7	2.5	virginica
+6.7	3.0	5.2	2.3	virginica
+6.3	2.5	5.0	1.9	virginica
+6.5	3.0	5.2	2.0	virginica
+6.2	3.4	5.4	2.3	virginica
+5.9	3.0	5.1	1.8	virginica
```
Comment (c21) [inline python/ray/data/examples/data/iris.tsv:150]: nit: one new line at the end.

**Review by c21 — COMMENTED** [review]
Thanks @pcmoritz! Looks good except several minor comments.

File: python/ray/data/read_api.py (hunk 3 above)
Comment (pcmoritz) [reply python/ray/data/read_api.py:568]: I tried that before and then it didn't render correctly in the docs, let me try again :)

File: python/ray/data/examples/data/iris.tsv (hunk 4 above)
Comment (pcmoritz) [reply python/ray/data/examples/data/iris.tsv:2]: Thanks, I fixed that now :)

File: python/ray/data/examples/data/iris.tsv (hunk 5 above)
Comment (pcmoritz) [reply python/ray/data/examples/data/iris.tsv]: fixed

File: python/ray/data/read_api.py (hunk 3 above)
Comment (pcmoritz) [reply python/ray/data/read_api.py]: Ok, so the right solution that makes this work in all cases (both raw docstring and readthedocs) seems to be to put an `r` before the doc string and not escape this, which I did now :)

**Comment by pcmoritz** [comment]
This should be ready to merge now, @c21 can you approve when you get a chance?

**Review by c21 — APPROVED** [review]
Thanks @pcmoritz, LGTM!
